### PR TITLE
Divert CMake specs to regular test builds

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Since cmake is a method for how the SDK is built,
  from a test perspective it is the same sdk. Therefore
  we will reuse the same test jobs as the non-cmake specs.

[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
cc @dnakamura @llxia .
I'd like to wait for #5787 in order to test this change.

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
@vsebe for review. The build fails as expected but the Test that is launched is `Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal` (no `_cm`)

**Comment by AdamBrousseau** [comment]
@pshipton as an fyi, the decision here is to use the same test build for both cmake and non-cmake specs. Which will interleave the history of the two "specs". Also note, at the moment we are not running any tests on the cmake spec.

**Comment by DanHeidinga** [comment]
> Which will interleave the history of the two "specs".

@AdamBrousseau Does this apply to more than the personal builds and pull request testing?

I would prefer to not intermingle nightly builds as they are used for build triage purposes

**Comment by AdamBrousseau** [comment]
It will apply to any/all builds that have a second spec for cmake.
When I discussed with @llxia originally, our 3 options were:
1. Add the Jenkins file to the test repo for the cmake spec(s)
1. Catch the cmake spec on our side and remove the `_cm`
1. Catch the cmake spec on test side and remove the `_cm`

We went with option 2 since as far as the test is concerned, there is no concept of a different spec for cmake. I'd have to look at the test job generation code (or talk to @llxia) to see the feasibility of a 4th option to generate a cmake job with a "regular" spec in order to avoid the history mixing.

**Comment by pshipton** [comment]
From a build monitoring perspective, I don't care if the tests are intermingled. It won't affect triage of failures, and any Issues created will refer to the appropriate link. What it will affect is the history, we'll effectively only have half the history on platforms that run both regular and cmake builds. If we are planning to move quickly to cmake once it is enabled, and stop using non-cmake builds then this won't be a problem.

**Comment by DanHeidinga** [comment]
> If we are planning to move quickly to cmake once testing is enabled, and stop using non-cmake builds then this won't be a problem.

The plan is to get cmake working platform by platform in jdk11 with platforms spaced 1-1.5 weeks apart.  Does that meet your definition of quickly?

**Comment by pshipton** [comment]
We have a 30 day history now, which changes to 15 when they are interleaved, which is more than 1.5 weeks. As we only had a 10 day history before yesterday when #6021 changed it, I don't see a problem.

**Comment by AdamBrousseau** [comment]
So to confirm :+1:  , we're ok with this change as-is.

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

File: buildenv/jenkins/common/pipeline-functions [hunk 1]
```diff
@@ -577,6 +578,7 @@ def generate_test_jobs(TARGET_NAMES, SPEC, ARTIFACTORY_SERVER, ARTIFACTORY_REPO)
     levels.unique(true)
     groups.unique(true)
 
+    SPEC = strip_cmake_specs(SPEC)
```
Comment (vsebe) [inline buildenv/jenkins/common/pipeline-functions]: SPEC is global variable, this overrides it.  Summary badge needs the original spec value.

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
@vsebe seems to be working now
https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK11_x86-64_linux_cm/238/console
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/508/

**Comment by AdamBrousseau** [comment]
@llxia for final review/merge.

File: buildenv/jenkins/common/pipeline-functions [hunk 2]
```diff
@@ -685,4 +687,13 @@ def setup_pull_request() {
     PERSONAL_BUILD = 'true'
 }
 
+def strip_cmake_specs(spec) {
+    // Strip off cmake from SPEC name
+    // Since cmake is a method for how the SDK is built,
+    // from a test perspective it is the same sdk. Therefore
+    // we will reuse the same test jobs as the non-cmake specs.
+    spec -= '_cm'
+    echo "Strip cmake spec:${spec} -> ${new_spec}"
```
Comment (llxia) [inline buildenv/jenkins/common/pipeline-functions]: Where is the `new_spec ` defined?

Comment (AdamBrousseau) [reply buildenv/jenkins/common/pipeline-functions]: Good catch, I had changed it slightly after the test build. I will re-test.

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

File: buildenv/jenkins/common/pipeline-functions (hunk 2 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/pipeline-functions]: Fixed @llxia

**Comment by llxia** [comment]
The correct test build is triggered. Test compilation failure is not related to this build pipeline script change.

**Pull request merged**